tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,221 @@
+const bracPlaceFactory = require('../../factories/brac/contacts/brac-place');
+const bracPersonFactory = require('../../factories/brac/contacts/brac-person');
+const bracUserFactory = require('../../factories/brac/users/brac-user');
+const bracSurvey = require('../../factories/brac/reports/brac-survey');
+const fs = require('fs');
+const path = require('path');
+const Faker = require('@faker-js/faker');
+const dataConfig = require('./data-config.json');
+const sizeConfig = require('./size-config.json');
+const createCsvWriter = require('csv-writer').createObjectCsvWriter;
+const args = process.argv.slice(2);
+//const [, , threadId, directory] = process.argv;
+const districtHospitalName = args[0] + '-districthospital';
+const numberOfDistrictHospitals = sizeConfig.number_of_district_hospitals;
+const numberOfManagersPerDistrictHospital = sizeConfig.number_of_managers_per_district_hospitals;
+const numberOfHealthCentersPerDistrictHospital = sizeConfig.number_of_health_centers_per_district_hospital;
+const numberOfChwPerHealthCenter = sizeConfig.number_of_chw_per_health_center;
+const numberOfClinicsPerHealthCenter = sizeConfig.number_of_clinics_per_health_center;
+const numberOfFamilyMembers = sizeConfig.number_of_family_members;
+const dataDirectory = args[1];
+const preconditionDirectory = dataConfig.precondition_data_directory;
+const mainDirectory = dataConfig.main_script_data_directory;
+const jsonDirectory = dataConfig.json_directory;
+const preconditionDataDirectory = path.join(
+  dataDirectory,
+  dataConfig.precondition_data_directory,
+  dataConfig.json_directory);
+const mainDataDirectory = path.join(dataDirectory, dataConfig.main_script_data_directory);
+const usersDirectory = path.join(dataDirectory, dataConfig.precondition_data_directory);
+const dataExtension = dataConfig.json_data_extension;
+const csvWriter = createCsvWriter({
+  path: path.join(usersDirectory, 'users.csv'),
+  header: [
+    { id: 'username', title: 'username' },
+    { id: 'password', title: 'password' },
+    { id: 'roles', title: 'roles' },
+    { id: 'contact', title: 'contact' },
+    { id: 'phone', title: 'phone' },
+    { id: 'place', title: 'place' },
+  ],
+  alwaysQuote: true,
+  append: true
+});
+
+const users = [];
+const managers = [];
+
+const createDataDoc = async (filePath, fileName, content) => {
+  try {
+    await fs.promises.writeFile(path.join(filePath, fileName), content);
+  } catch (err) {
+    console.error('CreateDataDoc ' + err);
+    throw err;
+  }
+};
+
+const createDataDirectory = async (directoryPath, directoryName) => {
+  try {
+    await fs.promises.mkdir(path.join(directoryPath, directoryName));
+  } catch (err) {
+    if (err.code === 'EEXIST') {
+      return;
+    }
+    console.error('CreateDataDirectory ' + directoryName + ' failed - ' + err);
+    throw err;
+  }
+};
+
+const setParents = (firstLevelId, secondLevelId, thirdLevelId) => {
```

Comment:
We typically call this "lineage", because you define parent, grandparent and greatgrandparent. To make it consisteny with the rest of our codebase, can you rename this to: `getLineage` ?